src/Modules/FreeModElem-orderings.jl

```diff
@@ -196,7 +201,11 @@ function Base.iterate(a::GeneralPermutedIterator{:monomials, <:FreeModElem{<:MPo
   state += 1
   state <= length(a.perm) || return nothing
   (i, j) = a.perm[state]
-  return monomial(a.elem[i], j)*parent(a.elem)[i], state
+  mij = monomial(a.elem[i], j)
+  F = parent(a.elem)
+  R = base_ring(F)
+  s = sparse_row(R, [i], [mij])
```

Comment:
The changes in this `iterate` method reduce allocations from 24 to 12. By adding the undocumented keyword arg `sort=false` we can reduce it further to 9.
```suggestion
  s = sparse_row(R, [i], [mij]; sort=false)
```

I'll update this PR further and will also submit an optimization to Hecke to make this unnecessary (basically: if the vectors have length 1 we don't need to sort them...